### PR TITLE
Add contributor and rewrite pobi.

### PR DESCRIPTION
**Pull request by tudor-malene**
draft

### CODE REVIEW

File: detailed-design/pobi.md [hunk 1]
```diff
@@ -1,86 +1,349 @@
...
+1. The aggregator that has a single rollup without competition included in a block that was generated with the proof of
+   the parent L1 block, will get the full reward. This is the most efficient case that is encouraged.
 
-The reward rules are depicted in the following diagram:
-![L1 front running](../images/block-rewarding.png)
+2. If multiple rollups generated with the same L1 proof and different nonces are published in the same block, the one
+   with the lowest nonce is on the canonical chain, but the reward is split between them in a proportion of 80/20 (this
+   ratio is indicative only). The reason for this rule is that it incentivises aggregators to gossip their winning
+   rollup such that no one else publishes at the same time.
+    - There is no incentive for the losing aggregator to publish as 20% of the reward will not even cover the cost of
```
Comment (jamescarlyle) [inline detailed-design/pobi.md]: Who says 20% won't cover gas? Doesn't it depend on the reward size and the gas price?

Comment (tudor-malene) [reply detailed-design/pobi.md]: Fair point. 
The assumption is that the rewards for the aggregators won't be that high, as we don't want to fleece the users.

**Review by jamescarlyle — APPROVED** [review]

**Comment by jamescarlyle** [comment]
@tudor-malene please can you merge this?

File: detailed-design/pobi.md [hunk 2]
```diff
@@ -1,84 +1,286 @@
...
+
+TODO - does it matter the block difference? a. _WRn_ is included in an earlier block _WRn_ is on the canonical chain
+
+b. _WRn_ is included in a later block.
+
+[comment]: <> ([TODO - diagrams])
+
+#### Only the rollup with the higher nonce is mined in the next L1 block.
+
+If the rollup is valid, then all aggregators will use it as the parent of the current rollup, basically it will be on the canonical chain. The reward
```
Comment (GavT) [inline detailed-design/pobi.md]: This line is unfinished

Comment (tudor-malene) [reply detailed-design/pobi.md]: oops

File: detailed-design/threat-model.md [hunk 3]
```diff
@@ -1,41 +1,100 @@
...
+#### Write-Level Attacks
 There are a few variants of this attack.
 
-The most severe is when the attacker impersonates a TEE and signs a rollup containing illegal transactions. If there is no withdrawal attempt in the rollup, this attack is equivalent to an L1 node publishing an invalid block. All the honest TEEs will ignore it. Even in this case, the protocol reverts to the behavior of a typical blockchain.
+The most severe is when the attacker impersonates a TEE and signs a rollup containing illegal transactions. If there is no withdrawal attempt in the rollup, or there are no other external side effects, this attack is equivalent to an L1 node publishing an invalid block. All the honest TEEs will ignore it. Even in this case, the protocol reverts to the behavior of a typical blockchain.
+
+If the attacker impersonating an enclave attempts to withdraw funds from the bridge contract based on invalid transactions, is a case that needs special consideration. This is where the L1 protocol meets the L2 protocol, and the bridge contract has to decide the validity of the withdrawal request in order to execute it. The solution is described in detail in the POBI protocol.
```
Comment (GavT) [inline detailed-design/threat-model.md]: Minor grammatical error: "invalid transactions, **this** is a case"

File: detailed-design/threat-model.md [hunk 4]
```diff
@@ -1,41 +1,100 @@
...
+
+#### Colluding Write Level Attacks
+This is an extreme attack on this protocol, where all aggregators find a way to break the TEE and are colluding to steal all funds.
+
+The defense is to have a reasonable number of active verifiers that have a stake in the good functioning of Obscuro. "Active" means that they are watching the Obscuro ledger in real time. These actors will detect if there is a malicious head rollup and no other valid fork is being aggregated.
```
Comment (GavT) [inline detailed-design/threat-model.md:37]: Not sure this needs to be added to the WP but we should not assume enough people / verifiers will actively watch for good behaviour. Most either won't care enough or will assume someone else is watching.

File: detailed-design/threat-model.md [hunk 5]
```diff
@@ -1,41 +1,100 @@
...
+
+
+#### Sybil attacks
+
+This section will analyze the threats that a powerful adversary able to create a large number of aggregators can pose on the protocol.
+The reasoning around this attack is quite different from typical public blockchains.
+
+There are two reasons for such an attack:
+- The attackers want to steal the funds locked in the L1 Bridge. Note that this is not a problem in typical L1 chains as there are no side effects there, but it is in L2s.
+- Double spending. The attacker plans a malicious re-org or the L2 ledger. This is the typical threat for sybil attacks in L1 networks.
```
Comment (GavT) [inline detailed-design/threat-model.md]: Typo: "re-org or the L2 ledger"

File: detailed-design/threat-model.md [hunk 6]
```diff
@@ -1,41 +1,100 @@
...
+If the attacker is not able to hack the TEE, they’re not able to deviate from the canonical chain or to insert illegal transactions, as the software won’t let them. 
+Having a majority on the Obscuro network will not help.
+
+To perform a double spend, the attackers have to actually perform a double spend attack on the L1 blocks themselves that contain the rollups. 
+
+
+###### Sybil attack when hacking the TEE
 
-There is one case that needs special consideration. When the attacker impersonating an enclave attempts to withdraw funds from the Bridge contract based on invalid transactions. This is where the L1 protocol meets the L2 protocol, and the Bridge contract has to decide the validity of the request. The solution is described in detail in the POBI protocol.
+In this scenario, the attacker is able to hack the TEE, and they setup a large number of aggregators, limited only by the stake.  Their motivation is that they want to steal the funds from the bridge – the ultimate prise.
```
Comment (GavT) [inline detailed-design/threat-model.md]: Typo: "prise" should be "prize".

File: detailed-design/pobi.md [hunk 7]
```diff
@@ -1,86 +1,286 @@
...
+- the aggregator who generated it
+- the generation of the L1 block used as proof
+- the generation of the L1 block that includes this rollup.
+- the nonce
+
+The notation is the following: _R_$Rollup_Generation[$Aggregator, $L1_Proof_Generation, $L1_Block_Generation, $Nonce]_
+
+Note that: L1_Proof_Generation < L1_Block_Generation .
+
+Example: _R_15[Alice, 100, 102, 20]_ means the generation is 15, the aggregator is _Alice_, the generation of the L1 bock used as proof is 100, the generation of the L1 bock that included the rollup is 102 , and the nonce=20
```
Comment (CaisManai) [inline detailed-design/pobi.md]: unnecessary space between 102 and the comma

File: detailed-design/pobi.md [hunk 8]
```diff
@@ -1,86 +1,286 @@
...
+
+### The canonical chain
+
+The POBI protocol allows any aggregator to publish rollups to the management contract. This means that short-lived forks are a normal part of the protocol. The ObscuroVM running inside the TEE of every node must be able to deterministically select one of the forks as the canonical chain and only append a rollup on top of that. This means that the TEE must receive all the relevant content of the L1 blocks, and the logic must be identical on all nodes.
+
+The rules for the canonical chain:
+
+1. The genesis rollup is part of the canonical chain, and will be included in a block by the first aggregator.
+
+2. If an L1 block contains a single rollup who's parent is the head rollup of the canonical chain included in a previous L1 block, it will also be on the canonical chain if no other rollup with the same parent was included in an earlier block. Any other sibling rollup included in a later block is not on the canonical chain. This is the _Primogeniture_ rule, where a rollup is born when included in an L1 block.
```
Comment (CaisManai) [inline detailed-design/pobi.md]: should be 'whose'

File: detailed-design/cryptoeconomics.md [hunk 9]
```diff
@@ -1,20 +1,20 @@
 ## Cryptoeconomics
-All successful decentralised solutions need a strong incentive mechanism to keep the protocol functioning efficiently. Obscuro also aims to give its users a good experience by not allowing any form of "MEV". This means it must avoid the gas price bidding war we constantly see in popular blockchains.
+All successful decentralised solutions need a strong incentive mechanism to keep the protocol functioning efficiently. On top of that, Obscuro also aims to give its users a good experience by not allowing any form of "MEV". This means it must avoid the gas price bidding war we constantly see in popular blockchains.
```
Comment (CaisManai) [inline detailed-design/cryptoeconomics.md]: should this be 'effectively' rather than 'efficiently'?

File: detailed-design/cryptoeconomics.md [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
 One disadvantage of this model is that fees can get very high in periods of network congestion, which degrades user experience.
 
 Ethereum builds on top of this model to handle the complexities of Turing complete smart contract execution, by introducing a notion of _gas_ and of _gas price_, but the high level mechanics remain the same.
 
-The incentive design for a decentralised L2 protocol must also consider the problem of front-running. For a rollup to be final, it has to be added to a L1 block, which is where a L1 miner or staker can attempt to claim the reward of a different L2 node.
+The incentive design for a decentralised L2 protocol must also consider the problem of front-running the actual rollup. For a rollup to be final, it has to be added to a L1 block, which is where a L1 miner or staker can attempt to claim the reward that rightfully belongs to a different L2 node.
```
Comment (CaisManai) [inline detailed-design/cryptoeconomics.md:17]: do we fully explain this idea of L1 miners attempting to claim rewards elsewhere?

File: detailed-design/threat-model.md (hunk 4 above)
Comment (tudor-malene) [reply detailed-design/threat-model.md:37]: That's a good point. But there will be someone watching, and that is the governance of Obscuro.
I hinted at that with: "that have a stake in the good functioning of Obscuro."
Should it be more explicit?

File: detailed-design/cryptoeconomics.md (hunk 10 above)
Comment (tudor-malene) [reply detailed-design/cryptoeconomics.md:17]: yes, below.
This paragraph just lists the high level problems. 
Should I make it more apparent?

File: detailed-design/pobi.md (hunk 2 above)
Comment (tudor-malene) [reply detailed-design/pobi.md]: Removed the entire section.

File: detailed-design/threat-model.md (hunk 4 above)
Comment (tudor-malene) [reply detailed-design/threat-model.md:37]: Added this:
_Note: One such actor monitoring the network will be the Obscuro Foundation, which has the mandate to keep the protocol functioning correctly. The protocol also rewards other independent parties to take on this job by assigning a small amount each rollup to a random validator if they can prove they are active._

**Pull request merged**